Domains: Check for privacy support before adding subscription to cart.

delputnam opened the pull request:
If a domain doesn't support a privacy subscription we shouldn't add it to the cart. This PR checks domains added to the cart in the signup flow and prevents them from being added if the domain doesn't support a privacy purchase.

Depends on D9363-code.

Build calypso locally and browse to:
calypso.localhost:3000/start/domain-first?new=asdfasdfasdfasdf.blog
Select "New Site".
Repeat, but select "Just Buy A Domaibn".
For both of these options, when you get to the checkout screen, you should have a privacy subscription in the cart.

Now browse to:
calypso.localhost:3000/start/domain-first?new=asdfasdfasdfasdf.blog
Select "New Site".
Repeat, but select "Just Buy A Domaibn".
Now you should now have a privacy subscription in the cart since .ca doesn't support purchasing privacy separately.

aidvu commented on client/lib/signup/step-actions.js:
```diff
@@ -150,17 +151,19 @@ export function createSiteWithCart(
-							source: 'signup',
-						} );
+				if ( supportsPrivacyPurchase( domainItem ) ) {
+					if ( domainItem ) {
```
Can do `domainItem && supportsPrivacyPurchase( domainItem )`.

aidvu commented on client/lib/signup/step-actions.js:
```diff
@@ -46,7 +47,7 @@ export function createSiteOrDomain( callback, dependencies, data, reduxStore ) {
 
 		const domainChoiceCart = [ domainItem ];
-		if ( domainItem ) {
+		if ( supportsPrivacyPurchase( domainItem ) ) {
```
This will error if no `domainItem`. :)

delputnam replied:
Yep. Thanks.  Will fix.

delputnam commented:
So, I think we came to the conclusion that this check is something that should be done on the front end rather than on the back end. If I have this wrong, please let me know.

I've updated the product flag to `is_privacy_protection_product_purchase_allowed` in this PR and on the back end and also changed the name of the function that checks for the value of this flag to `supportsPrivacyProtectionPurchase` in order to be more internally consistent with naming for this feature.
